lib/cretonne/build.rs

```diff
@@ -44,3 +46,55 @@ fn main() {
         process::exit(status.code().unwrap());
     }
 }
+
+fn output_target_cfgs() {
+    let config = env::var("CRETONNE_TARGETS").unwrap_or("all".into());
+    let possible_isas = ["riscv", "intel", "arm32", "arm64"];
+
+    match config.as_ref() {
+        "all" => {
+            for isa in &possible_isas {
+                println!("cargo:rustc-cfg=build_{}", isa);
+            }
+        }
+
+        "native" => {
+            let target = env::var("TARGET").expect("The TARGET environment variable must be set");
+            let arch = target.split("-").next().unwrap();
+
+            let isa = match arch {
+                "aarch64" => "arm64",
+                "arm" | "armv7" | "armv7s" => "arm32",
+                "x86_64" | "i386" | "i586" | "i686" => "intel",
+                "riscv" => "riscv",
+                _ => {
+                    println!("no supported isa found for target triple `{}`", target);
+                    process::exit(1);
```

Comment:
Did you check that this message actually appears on screen? Cargo normally swallows the build script output. Should it be written to stderr instead?